File: loginandsignuppage/app/src/main/java/com/example/loginandsignuppage/MainActivity.java
package com.example.loginandsignuppage;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

EditText LoginText,PasswordText;
Button btnlogin;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        LoginText=(EditText) findViewById(R.id.login);
        PasswordText=(EditText) findViewById(R.id.Password);

        btnlogin=(Button) findViewById(R.id.buttonlogin);

        btnlogin.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String username = LoginText.getText().toString();
                String Password = PasswordText.getText().toString();

                if (username.equals("vinay") && (Password.equals("12345"))) {
                    Toast.makeText(MainActivity.this, "welcome", Toast.LENGTH_SHORT).show();
                    Intent intent = new Intent(getApplicationContext(), NewActivity.class);
                    startActivity(intent);
                } else {
                    Toast.makeText(MainActivity.this, "invalid credentials", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: KtAlertDialog/app/src/main/java/com/example/ktalertdialog/MainActivity.kt
package com.example.ktalertdialog

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AlertDialog

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val button=findViewById<Button>(R.id.button)

        button.setOnClickListener {
            val builder=AlertDialog.Builder(this)    //set title for alert dialog
            builder.setTitle(R.string.title)              //set title for alert dialog
            builder.setMessage(R.string.message)    //set message for alert dialog

            builder.setPositiveButton("yes"){dialogInterface,which-> //performing positive action
                Toast.makeText(applicationContext,"clicked yes",Toast.LENGTH_SHORT).show()
            }
            builder.setNegativeButton("no"){dialogInterface,which->//performing negative action
                Toast.makeText(applicationContext,"clicked no",Toast.LENGTH_SHORT).show()
            }
            builder.setNeutralButton("Cancle"){dialoginterface,which->//performing cancel action
                Toast.makeText(applicationContext,"clicked cancle",Toast.LENGTH_SHORT).show()
            }
            val alertDialog:AlertDialog=builder.create()//Create AlretDialog
            alertDialog.setCancelable(false)//set other dialog properties
            alertDialog.show()
        }


    }
}